scripts/main/header.js

```diff
@@ -250,41 +347,100 @@ header.setMode = function(mode) {
 			header.dom('.header__toolbar--public, .header__toolbar--albums, .header__toolbar--album, .header__toolbar--map').removeClass('header__toolbar--visible');
 			header.dom('.header__toolbar--photo').addClass('header__toolbar--visible');
 
+			tabindex.makeFocusable(header.dom('.header__toolbar--photo'));
+			tabindex.makeUnfocusable(header.dom('.header__toolbar--public, .header__toolbar--albums, .header__toolbar--album, .header__toolbar--map'));
 			// If map is disabled, we should hide the icon
 			if (lychee.publicMode===true ? lychee.map_display_public : lychee.map_display) {
-				$('#button_map').show();
+				e = $('#button_map');
+				e.show();
+				tabindex.makeFocusable(e);
 			} else {
-				$('#button_map').hide();
+				e = $('#button_map');
+				e.hide();
+				tabindex.makeUnfocusable(e);
 			}
 
 			if (album.isUploadable()) {
-				$('#button_trash, #button_move, #button_visibility, #button_star').show()
+				e = $('#button_trash, #button_move, #button_visibility, #button_star');
+				e.show();
+				tabindex.makeFocusable(e);
 			} else {
-				$('#button_trash, #button_move, #button_visibility, #button_star').hide();
+				e = $('#button_trash, #button_move, #button_visibility, #button_star');
+				e.hide();
+				tabindex.makeUnfocusable(e);
 			}
 
 			if (photo.json && photo.json.hasOwnProperty('share_button_visible') && photo.json.share_button_visible !== '1') {
-				$('#button_share').hide();
+				e = $('#button_share');
+				e.hide();
+				tabindex.makeUnfocusable(e);
 			} else {
-				$('#button_share').show();
+				e = $('#button_share');
+				e.show();
+				tabindex.makeFocusable(e);
 			}
 
 			// Hide More menu if empty (see contextMenu.photoMore)
 			$('#button_more').show();
+			tabindex.makeFocusable($('#button_more'));
 			if (!(album.isUploadable() ||
 				(photo.json.hasOwnProperty('downloadable') ? photo.json.downloadable === '1' :
 				album.json && album.json.downloadable && album.json.downloadable === '1')) &&
 				!(photo.json.url && photo.json.url !== '')) {
-				$('#button_more').hide();
+				e = $('#button_more');
+				e.hide();
+				tabindex.makeUnfocusable(e);
 			}
 
+			// Remove buttons if needed
+			if(!lychee.enable_button_visibility) {
+				e = $('#button_visibility', '.header__toolbar--photo');
+				e.hide();
```

Comment:
Can't you use https://api.jquery.com/remove/ instead of `.hide()` ?

I feel this would simplify the dom. :)